src/iuse_actor.cpp

```diff
@@ -2793,17 +2793,19 @@ repair_item_actor::repair_type repair_item_actor::default_action( const item &fi
     return RT_NOTHING;
 }
 
-bool damage_item( player &pl, item &fix )
+bool damage_item( player &pl, item_location &fix )
 {
-    pl.add_msg_if_player( m_bad, _( "You damage your %s!" ), fix.tname().c_str() );
-    if( fix.inc_damage() ) {
+    pl.add_msg_if_player( m_bad, _( "You damage your %s!" ), fix->tname() );
+    if( fix->inc_damage() ) {
         pl.add_msg_if_player( m_bad, _( "You destroy it!" ) );
-        const int pos = pl.get_item_position( &fix );
-        if( pos != INT_MIN ) {
+        if( fix.where() == item_location::type::character ) {
+            const int pos = pl.get_item_position( fix.get_item() );
             pl.i_rem_keep_contents( pos );
         } else {
-            // NOTE: Repairing items outside inventory is NOT yet supported!
-            debugmsg( "Tried to remove an item that doesn't exist" );
+            for( auto &content : fix->contents ) {
+                g->m.add_item_or_charges( fix.position(), content );
```

Comment:
What if it's in a vehicle?

Reply:
yikes, it looks like that complicates things a fair bit! I've pushed up a change which refactors this into a new method on `item_location` called `remove_item_keep_contents`, since it looks like there's no way to access the vehicle information for an `item_location` that's in a vehicle externally to the `item_location::impl`. Let me know if that's not the right way to go.

Reply:
That definitely works.  I probably would have just dropped the contents in the vehicle cargo or on to the map with `put_into_vehicle_or_drop()` then removed the item, but it's  nice to have an equivalent to `character::i_rem_keep_contents()`.

Reply:
Hm, yeah I thought about that. It might be better, actually, because it would properly print messages, and wouldn't duplicate the dropping code. The `remove_item_keep_contents` method I've added loses the reference to the character that's doing the dropping, so it can't print the appropriate messages.

Downside is that the item would be dropped where the player is rather than where the item is. Which might be fine, actually?

Reply:
>Hm, yeah I thought about that. It might be better, actually, because it would properly print messages, and wouldn't duplicate the dropping code. The remove_item_keep_contents method I've added loses the reference to the character that's doing the dropping, so it can't print the appropriate messages.

Right I forgot about printing messages.  Your call I say, I don't have a strong opinion about it either way.

>Downside is that the item would be dropped where the player is rather than where the item is. Which might be fine, actually?

You can pass a location to `put_into_vehicle_or_drop()`, and you can get that location from the `item_location`

Reply:
i've undone the refactor and now am using `put_into_vehicle_or_drop`.